### PR TITLE
Connects to AgileVentures/MetPlus_tracker#133

### PR DESCRIPTION
**Pull request by patmbolger**
also,

Connects to AgileVentures/MetPlus_tracker#186

This is 2nd of 3 phases for these stories.  This add notification emails (delivered in blocking mode - 3rd phase will be background/async delivery).

### CODE REVIEW

File: app/mailers/agency_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class AgencyMailer < ApplicationMailer
+
+  def job_seeker_registered(agency_emails, js_name, js_id)
```
Comment (joaopapereira) [inline app/mailers/agency_mailer.rb]: Wouldnt it be best to pass the JS object instead of parts of it?

File: app/mailers/agency_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    send_notification_mail(agency_emails, js_name, js_id, 'Job Seeker')
+  end
+
+  def company_registered(agency_emails, company_name, company_id)
```
Comment (joaopapereira) [inline app/mailers/agency_mailer.rb]: Wouldnt it be best to pass the Company object instead of parts of it?

File: app/models/event.rb [hunk 3]
```diff
@@ -9,9 +10,20 @@ def self.create(evt_type, evt_data)
     when :JS_REGISTER     # job seeker registered
       Pusher.trigger('pusher_control',
                      EVT_TYPE[:JS_REGISTER], evt_data)
+
+      # Send notification email to all agency personnel
+      agency_emails = AgencyPerson.where(agency: Agency.first).pluck(:email)
```
Comment (joaopapereira) [inline app/models/event.rb]: Shouldn't this logic be inside the Agency to retrieve all the emails from AgencyPerson?

File: app/mailers/agency_mailer.rb (hunk 1 above)
Comment (joaopapereira) [inline app/mailers/agency_mailer.rb]: Hmm I see you dont have the object there, shouldn't we get it and use it?

**Comment by joaopapereira** [comment]
Generally looks good, please review my comments

File: app/mailers/agency_mailer.rb (hunk 1 above)
Comment (patmbolger) [inline app/mailers/agency_mailer.rb]: Good question.  I designed *Event.create* to take two arguments: 1) a key for which event is to be created and 2) a task containing attributes needed for both Pusher and email notifications.  This allows us to add other events that may not be associated with a distinct object that contains all the attributes needed for notification.

For these first two events, I could certainly pass in the object (JS, Company) and be fine.  But other events may have different data attributes that don't lend themselves to that model.  So, a hash of attributes seems more generic for growth.

File: app/models/event.rb (hunk 3 above)
Comment (patmbolger) [inline app/models/event.rb]: You're right .... my head must have still been in prototype mode when I wrote that

File: app/models/agency.rb [hunk 4]
```diff
@@ -10,27 +10,31 @@ class Agency < ActiveRecord::Base
...
     raise RuntimeError, 'Logged in user is not an agency person' unless
             user.actable.is_a? AgencyPerson
-            
+
     user.actable.agency
   end
-  
+
+  def self.all_agency_people_emails
+    first.agency_people.pluck(:email)
```
Comment (joaopapereira) [inline app/models/agency.rb]: This need to be changed in the future when we have multiple agencies, please add a comment or something similar

**Comment by joaopapereira** [comment]
Please just review my latest commit and I believe we are ready for merge, unless anyone else have some question

**Comment by joaopapereira** [comment]
Looks good going to merge

**Pull request merged**